frontend/src/metabase/dashboard/components/parameters/widgets/TextWidget.jsx

```diff
@@ -32,7 +35,15 @@ export default class TextWidget extends Component {
     }
 
     render() {
-        const { setValue, className, isEditing } = this.props;
+        const { setValue, className, isEditing, focusChanged: parentFocusChanged } = this.props;
+        const defaultPlaceholder = this.state.isFocused ? "" : (this.props.placeholder || "Enter a value...");
+        const self = this;
+
+        function focusChanged(isFocused) {
+            if (parentFocusChanged) parentFocusChanged(isFocused);
+            self.setState({isFocused})
+        }
+
         return (
             <input
```

Comment:
I think we could make this `<input>` have a transparent background so it looks better in full screen etc